### PR TITLE
Rc 36667 : Conjurer TTL Support for terraform

### PR DESCRIPTION
**Pull request by venkatm-rafay**

### CODE REVIEW

File: internal/resource_mks_cluster/mks_cluster_resource_gen.go [hunk 1]
```diff
@@ -1008,11 +1013,19 @@ func (v MetadataValue) String() string {
 func (v MetadataValue) ToObjectValue(ctx context.Context) (basetypes.ObjectValue, diag.Diagnostics) {
 	var diags diag.Diagnostics
 
-	annotationsVal, d := types.MapValue(types.StringType, v.Annotations.Elements())
```
Comment (chaitanya-rafay) [inline internal/resource_mks_cluster/mks_cluster_resource_gen.go:1011]: @venkatm-rafay Why are we changing annotations code here ?

Comment (venkatm-rafay) [reply internal/resource_mks_cluster/mks_cluster_resource_gen.go:1011]: @chaitanya-rafay These are the generated files. I din't make made any changes.

Comment (chaitanya-rafay) [reply internal/resource_mks_cluster/mks_cluster_resource_gen.go:1011]: @vasu-rafay Can you check this ?

Comment (vasu-rafay) [reply internal/resource_mks_cluster/mks_cluster_resource_gen.go:1011]: @chaitanya-rafay Looks good. Additional logic to handle `Unknown` and `Null` values are introduced for `Map`. @venkatm-rafay Can we add `terraform-plugin-codegen-framework` to go dependencies ?

File: rafay/resource_breakglassaccess.go [hunk 2]
```diff
@@ -272,7 +272,7 @@ func expandGroupExpiry(p []interface{}) []*systempb.GroupExpiryDetails {
 		ge := &systempb.GroupExpiryDetails{}
 
 		if v, ok := expiryMap["expiry"].(int); ok {
-			ge.Expiry = float32(v)
+			ge.Expiry = float64(v)
```
Comment (vasu-rafay) [inline rafay/resource_breakglassaccess.go:275]: @venkatm-rafay Is this required?

**Review by vasu-rafay — CHANGES_REQUESTED** [review]

File: rafay/resource_breakglassaccess.go (hunk 2 above)
Comment (venkatm-rafay) [reply rafay/resource_breakglassaccess.go:275]: yes @vasu-rafay it's throwing the error while doing make build. The field is of type Float64.

File: docs/data-sources/mks_cluster.md [hunk 3]
```diff
@@ -78,6 +78,7 @@ Read-Only:
 - `high_availability` (Boolean) Select this option for highly available control plane. Minimum three control plane nodes are required
 - `kubernetes_upgrade` (Attributes) (see [below for nested schema](#nestedatt--spec--config--kubernetes_upgrade))
 - `kubernetes_version` (String) Kubernetes version of the Control Plane
+- `installerTtl` (Integer) By default, this setting allows ttl configuration for installer config. If not provided by default will set ttl to 365 days.
```
Comment (vasu-rafay) [inline docs/data-sources/mks_cluster.md:81]: @venkatm-rafay it should be `installer_ttl`. TF schema follows underscore convention

File: docs/resources/mks_cluster.md [hunk 4]
```diff
@@ -26,6 +26,7 @@ resource "rafay_mks_cluster" "mks-noha-converged-cluster" {
     config = {
       auto_approve_nodes      = true
       kubernetes_version      = "v1.28.9"
+      installerTtl            = 365
```
Comment (vasu-rafay) [inline docs/resources/mks_cluster.md:29]: here as well

File: docs/resources/mks_cluster.md [hunk 5]
```diff
@@ -324,6 +327,7 @@ You can change the current Kubernetes version under `spec.config.kubernetes_vers
 
 **Optional**
 
+- `installerTtl` (Integer) By default, this setting allows ttl configuration for installer config. If not provided by default will set ttl to 365 days.
```
Comment (vasu-rafay) [inline docs/resources/mks_cluster.md:330]: here as well

**Review by vasu-rafay — CHANGES_REQUESTED** [review]

File: docs/resources/mks_cluster.md (hunk 4 above)
Comment (venkatm-rafay) [reply docs/resources/mks_cluster.md]: Done.

File: docs/resources/mks_cluster.md (hunk 5 above)
Comment (venkatm-rafay) [reply docs/resources/mks_cluster.md]: DOne

**Review by vasu-rafay — APPROVED** [review]

**Pull request merged**